mobile/lib/modules/backup/providers/backup.provider.dart

```diff
@@ -455,14 +463,27 @@ class BackupNotifier extends StateNotifier<BackUpState> {
     );
   }
 
-  void _onAssetUploaded(String deviceAssetId, String deviceId) {
-    state = state.copyWith(
-      selectedAlbumsBackupAssetsIds: {
-        ...state.selectedAlbumsBackupAssetsIds,
-        deviceAssetId
-      },
-      allAssetsInDatabase: [...state.allAssetsInDatabase, deviceAssetId],
-    );
+  void _onAssetUploaded(
+    String deviceAssetId,
+    String deviceId,
+    bool isDuplicated,
+  ) {
+    if (isDuplicated) {
+      _backupService.saveDuplicatedAssetIdToLocalStorage(deviceAssetId);
+      state = state.copyWith(
```

Comment:
doing this expensive update of the state involving rebuilding lists/sets every time an asset is uploaded, might noticeably slow down the backup if `allUniqueAssets` is large (10000+ assets).
I suggest to only update the state either after the backup is done or every n assets